### ISSUE TITLE
MM Profile updates leading to ResponseException: received 500 HTTP response

### ISSUE THREAD

**Issue by pdpinch** [mitodl/open-discussions#286]
I created a channel in production with the query https://micromasters.mit.edu/learners/?company_name[0]=F.%20A.%20Day%20Middle%20School&q=emmazp which should have subscribed only two learners, plus SCM staff. 

Every six hours, this channel raises ~800 `ResponseException` errors and a similar number of `JSONDecodeError` errors. 

The latter error suggests a configuration issue to me, but I'm also wondering why we are generating so many `DELETE` requests to the channel API. 

https://sentry.io/mit-office-of-digital-learning/discussions/issues/357589258/

```
ResponseException: received 500 HTTP response
(19 additional frame(s) were not displayed)
...
  File "rest_framework/views.py", line 477, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 437, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 474, in dispatch
    response = handler(request, *args, **kwargs)

ResponseException: received 500 HTTP response
```

**Comment by pdpinch** [comment]
https://sentry.io/mit-office-of-digital-learning/discussions/issues/357971728/

**Comment by noisecapella** [comment]
Every time a profile (or some program enrollment information, like grades) is updated we need to proactively update its membership. Since we don't have a membership list on open-discussions or micromasters we can only make requests to remove contributors, to make sure they are not contributors, and the same for subscribers.

**Comment by rhysyngsun** [comment]
This will probably go away with mitodl/micromasters#3661

**Comment by pdpinch** [comment]
fixed by mitodl/micromasters#3661
